cfripper/rules/base_rules.py

```diff
@@ -122,3 +142,28 @@ def valid_principals(self) -> Set[str]:
             if self._config.aws_account_id:
                 self._valid_principals.add(self._config.aws_account_id)
         return self._valid_principals
+
+
+class BaseDangerousPolicyActions(ResourceSpecificRule, ABC):
+    """
+    Base class for dangerous actions
+    """
+
+    REASON = "Resource {} should not be include the following dangerous actions: {}"
+    RISK_VALUE = RuleRisk.HIGH
+    GRANULARITY = RuleGranularity.ACTION
+
+    DANGEROUS_ACTIONS: List[str]
+
+    def resource_invoke(self, resource: Resource, logical_id: str) -> Result:
+        result = Result()
+        for policy in resource.policy_documents:
+            actions = policy.policy_document.get_allowed_actions()
+            dangerous_actions = set(actions) & set(self.DANGEROUS_ACTIONS)
+            self.add_failure_to_result(
```

Comment:
Should we check if dangerous_actions is empty?

Reply:
Yup, not a bad idea

Reply:
Done